src/dbally/context/context.py

```diff
@@ -0,0 +1,59 @@
+import ast
+from typing import Iterable
+
+from pydantic import BaseModel
+from typing_extensions import Self, TypeAlias
+
+from dbally.context.exceptions import ContextNotAvailableError
+
+# CustomContext = TypeVar('CustomContext', bound='BaseCallerContext', covariant=True)
+CustomContext: TypeAlias = "BaseCallerContext"
+
+
+class BaseCallerContext(BaseModel):
```

Comment:
pydantic is not a required dependency at the moment, and we should avoid it if possible (v1 vs v2 issues). Maybe a bare dataclass is enough here?

Reply:
We can still support pydantic models as context, but we shouldn't enforce it

Reply:
Fixed, thanks :)